File: Python3/0014_longest_common_prefix.py
# https://leetcode-cn.com/problems/longest-common-prefix/

class Solution: 
    def longestCommonPrefix(self, strs:List[str]) -> str:
        if strs == []:
            return ''
        
        nums = []
        for i in range(len(strs)):
            nums.append(len(strs[i]))
        min_len = min(nums)
        if min_len == 0:
            return ''
        
        for i in range(min_len):
            tmp = strs[0][i]
            for j in range(1,len(strs)):
                if tmp != strs[j][i]:
                    return strs[0][:i]
        return strs[0][:i+1]

File: Cpp/0014_longest_common_prefix.cpp
// https://leetcode-cn.com/problems/longest-common-prefix/

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string first = strs[0];
        int len_first = first.size();
        int len_strs = strs.size();

        for (int i = 0; i < len_first; ++i) {
            char c = first[i];

            for (int j = 1; j < len_strs; ++j) {
                if ((i == strs[j].size()) || (c != strs[j][i])) {
                    return first.substr(0, i);
                }
            }
        }
        return first;
    }
};